gammapy/cube/fit.py

```diff
@@ -346,6 +346,49 @@ def create(
             gti=gti,
         )
 
+    def stack(self, other):
+        """
+        Stack the maps in other onto self.
+        The stacking is done inplace
+
+        Parameters
+        ----------
+        other: `~gammapy.cube.MapDataset`
+            the MapDatasets to be stacked with this one.
+
+        Returns
+        -------
+        new : `~gammapy.cube.MapDataset`
+            A dataset with the stacked maps
+
+        """
+
+        if self.counts:
+            if other.mask_fit:
+                other.counts = other.counts * other.mask_fit
```

Comment:
I think we should not modify the `other` object when stacking. For the "in place" stacking we should only modify `self`. My proposal would be to add a `weights` argument (just as a numpy array) to `Map.coadd()`, which is multiplied to the values stacked in the map. So the code here would be:
```python3
self.counts.coadd(other.counts, weights=other.mask_safe)
```
Also note that we want to use `mask_safe` for the stacking not `mask_fit`.